workspace/__lib__/xod-dev/st7735-display/st7735-128x160/patch.cpp

```diff
@@ -0,0 +1,35 @@
+
+#pragma XOD error_raise enable
+
+struct State {
+    uint8_t mem[sizeof(ST7735)];
+};
+
+using Type = ST7735*;
+
+// clang-format off
+{{ GENERATED_CODE }}
+// clang-format on
+
+void evaluate(Context ctx) {
+    if (!isSettingUp())
+        return;
+
+    auto state = getState(ctx);
+
+    uint8_t cs = getValue<input_CS>(ctx);
+    uint8_t dc = getValue<input_DC>(ctx);
+    uint8_t rst = getValue<input_RST>(ctx);
+
+    if (!isValidDigitalPort(cs) || !isValidDigitalPort(dc) || !isValidDigitalPort(rst)) {
+        raiseError(ctx);
+        return;
+    }
```

Comment:
These checks can now be performed in compile-time. See `digital-write` for the reference.